### PR TITLE
Feature: async source

### PR DESCRIPTION
**Pull request by goern**

### CODE REVIEW

File: thoth/python/aiosource.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+        _LOGGER.debug("Versions available on %r (index with name %r): %r", self.url, self.name, result)
+        return result
+
+    async def _simple_repository_list_artifacts(self, package_name: str) -> AsyncIterableArtifacts:
+        """Parse simple repository package listing (HTML) and return artifacts present there."""
+        url = self.url + "/" + package_name
+        links = []
+
+        _LOGGER.debug(f"Discovering package %r artifacts from %r", package_name, url)
```
Comment (fridex) [inline thoth/python/aiosource.py]: No need for f-string.

File: thoth/python/aiosource.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            package_name = package_parts[-1]
+
+            # Discard links to parent dirs (package name of URL does not match the text.
+            link_text = link.text
+            if link_text.endswith("/"):
+                # Link names should end with trailing / according to PEEP:
+                #   https://www.python.org/dev/peps/pep-0503/
+                link_text = link_text[:-1]
+
+            if self.is_normalized_python_package_name(package_name) and package_name == link_text:
```
Comment (fridex) [inline thoth/python/aiosource.py]: `link_text` should be normalized, see https://github.com/thoth-station/python/pull/161

Comment (goern) [reply thoth/python/aiosource.py]: syn?

Comment (fridex) [reply thoth/python/aiosource.py]: Any reason why not always operate on normalized package names?

**Comment by goern** [comment]
regate

**Pull request merged**